src/Microsoft.Azure.WebJobs.Host/Scale/ScaleMonitorService.cs

```diff
@@ -0,0 +1,150 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Hosting;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Host.Scale
+{
+    /// <summary>
+    /// Service responsible for taking periodic scale metrics samples and persisting them.
+    /// </summary>
+    internal class ScaleMonitorService : IHostedService, IDisposable
+    {
+        private readonly Timer _timer;
+        private readonly IScaleManager _scaleManager;
+        private readonly IScaleMetricsRepository _metricsRepository;
+        private readonly ILogger _logger;
+        private readonly IOptions<ScaleOptions> _scaleOptions;
+        private readonly IPrimaryHostStateProvider _primaryHostStateProvider;
+        private bool _disposed;
+
+        public ScaleMonitorService(IScaleManager scaleManager, IScaleMetricsRepository metricsRepository, IOptions<ScaleOptions> scaleOptions, IPrimaryHostStateProvider primaryHostStateProvider, ILoggerFactory loggerFactory)
+        {
+            _scaleManager = scaleManager;
+            _metricsRepository = metricsRepository;
+            _timer = new Timer(OnTimer, null, Timeout.Infinite, Timeout.Infinite);
+            _logger = loggerFactory.CreateLogger<ScaleMonitorService>();
+            _scaleOptions = scaleOptions;
+            _primaryHostStateProvider = primaryHostStateProvider;
+        }
+
+        public virtual Task StartAsync(CancellationToken cancellationToken)
+        {
+            // start the timer by setting the due time
+            string tbsState = _scaleOptions.Value.IsTargetScalingEnabled ? "enabled" : "disabled";
```

Comment:
I don't understand the changes made to this service when it was migrated over. The existing code checks if RTS is enabled and only starts if it is. You've removed that logic why?

Reply:
2 reasons for that:
1. There is alrady RDS check in current code in DI path. So we check the same twice:
https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/ScriptHostBuilderExtensions.cs#L331
2. web jobs sdk should not readt RDS setting as it's function host stuff.